examples/factors/README.md

```diff
@@ -1,22 +1,203 @@
 ```
-cargo run --bin factors
+cd factors
+```
+
+## Step 2: Give the guest program a name
+
+This project will call a program that executes on the guest zkVM.
+It's currently named `method_name`.
+We want to name it something that represents what the guest program does -- let's call it `multiply`.
+Edit `methods/guest/Cargo.toml`, changing the line `name = "method_name"` to instead read
+```
+name = "multiply"
+```
+
+In order to access this guest code from the host driver program, the host program `factors/host/src/main.rs` includes two guest methods:
+
+```rust
+use methods::{METHOD_NAME_ELF, METHOD_NAME_ID};
+```
+
+Both of these must be changed to reflect the new guest program name:
+```rust
+use methods::{MULTIPLY_ELF, MULTIPLY_ID};
+```
+(As an aside, if you add more than one callable guest program to your next RISC Zero zkVM project, you'll need to include these `ELF` and `ID` references once for each guest file.)
+
+While we're at it, let's change the rest of the references in `factors/host/src/main.rs`.
+Don't worry about why these lines are included yet; for now, we're just being diligent not to leave dead references behind.
+Here are what the other two lines with `METHOD_NAME_ELF` and `METHOD_NAME_ID` should look like after updating:
+
+```rust
+    let mut prover =
+        Prover::new(MULTIPLY_ELF).expect("Prover should be constructed from valid ELF binary");
+
+...
+
+    receipt.verify(&MULTIPLY_ID).expect(
+        "Code you have proven should successfully verify; did you specify the correct image ID?",
+    );
+```
+
+### Intermission: Build and run the project!
+
+In the main project folder, build and run the project using `cargo run --release`.
+Nothing exciting will happen yet, but it should give you an error if you made one of the above changes incorrectly.
+Use this command any time you'd like to check your progress.
+
+## Step 3 (Host): Share two values with the guest
+
+In this step, we'll be continuing to modify `factors/src/main.rs`.
+Let's start by picking some aesthetically pleasing primes:
+```
+fn main() {
+    let a: u64 = 17;
+    let b: u64 = 23;
+```
+
+Currently, our host driver program creates and runs a prover.
+The `prover.run()` command will cause our guest program to execute:
+
+```rust
+    // Make the prover.
+    let mut prover =
+        Prover::new(MULTIPLY_ELF).expect("Prover should be constructed from valid ELF binary");
+
+    // TODO: Implement communication with the guest here
+
+    // Run prover & generate receipt
+    let receipt = prover.run().expect(
+        "Code should be provable unless it had an error or exceeded the maximum cycle limit",
+    );
+```
+ We'd like the host to make the values of `a` and `b` available to the guest prior to execution.
+ Because the prover is responsible for managing guest-readable memory, we need to share them after the prover is created.
+ To accomplish this, let's send our two values to the guest between the lines listed above:
```

Comment:
"between the lines listed above" feels easy to misread. 

Perhaps: "in place of the TODO in the previous code snippet" or "in between making the prover and running the prover."